external_parser/unit_tests/test_vw_external_parser.cc

```diff
@@ -73,4 +74,89 @@ BOOST_AUTO_TEST_CASE(cb_dedup_compressed) {
 
   clear_examples(examples, vw);
   VW::finish(*vw);
+}
+
+BOOST_AUTO_TEST_CASE(compare_dsjson_with_fb_models) {
+  std::string input_files = get_test_files_location();
+
+  std::string fb_model = input_files + "/test_outputs/m_average_fb.model";
+  std::string dsjson_model =
+      input_files + "/test_outputs/m_average_dsjson.model";
+
+  std::remove(fb_model.c_str());
+  std::remove(dsjson_model.c_str());
+
+  {
+    // run with flatbuffer joined logs
+    auto full_file_name =
+        input_files + "/valid_joined_logs/average_reward_100_interactions.fb";
+
+    auto vw = VW::initialize("--cb_explore_adf --binary_parser --quiet -f " + fb_model +
+                                 " -d " + full_file_name,
+                             nullptr, false, nullptr, nullptr);
+
+    v_array<example *> examples;
+    examples.push_back(&VW::get_unused_example(vw));
+    multi_ex multi_exs;
+    bool read_payload = false;
+    while (vw->example_parser->reader(vw, examples) > 0) {
```

Comment:
could this logic be replaced by calls similar to this https://github.com/VowpalWabbit/vowpal_wabbit/blob/b5939a84b16d27dfe6ab9cec7360be415899026f/python/pylibvw.cc#L264

Reply:
yeah probably, let me try (thanks!)

Reply:
yeah works fine